File: src/python/laboneq/core/types/enums/repetition_mode.py
# Copyright 2022 Zurich Instruments AG
# SPDX-License-Identifier: Apache-2.0

from enum import Enum


class RepetitionMode(Enum):
    """Controls how the repetition interval between averaging loop iterations is determined."""

    FASTEST = "fastest"
    """Execute iterations as fast as the hardware permits."""

    CONSTANT = "constant"
    """Execute at a fixed repetition time specified by the user."""

    AUTO = "auto"
    """Execute at a constant rate determined automatically from the longest iteration.

    Incompatible with ``SectionTimingMode.STRICT``.
    """

File: src/rust/laboneq-dsl/src/types/repetition_mode.rs
// Copyright 2026 Zurich Instruments AG
// SPDX-License-Identifier: Apache-2.0

use laboneq_units::duration::{Duration, Second};

#[derive(Debug, Clone, PartialEq, Copy)]
pub enum RepetitionMode {
    Fastest,
    Constant { time: Duration<Second, f64> },
    Auto,
}
